src/compositions/useTaskRunResult.ts

```diff
@@ -0,0 +1,45 @@
+import { UseSubscription, useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
+import { ComputedRef, computed, ref } from 'vue'
+import { useCan } from '@/compositions/useCan'
+import { useWorkspaceApi } from '@/compositions/useWorkspaceApi'
+import { Artifact, ArtifactsFilter } from '@/models'
+import { WorkspaceArtifactsApi } from '@/services/WorkspaceArtifactsApi'
+import { MaybeRef } from '@/types'
+
+export type UseTaskRunResult = {
+  subscription: UseSubscription<WorkspaceArtifactsApi['getArtifacts']>,
+  result: ComputedRef<Artifact | undefined>,
+}
+
+export function useTaskRunResult(taskRunId: MaybeRef<string | null | undefined>): UseTaskRunResult {
```

Comment:
should this use the watch pattern you used to account for reactive filters?

Reply:
I don't think we need to unless there's a performance consideration there; this is a primitive ref so changes should be adequately captured by the `computed`

Reply:
oh right, this doesn't take a filter. Ignore me.